draco/analysis/flagging.py

```diff
@@ -1592,6 +1637,71 @@ def mask_2d(self, y, w):
         # Flag times and frequencies that deviate by more than some threshold
         return dy > self.nsigma_2d
 
+    def mask_2d_sumthreshold(self, y, w):
+        """Iterative application of sumthreshold algorithm to mask large chi-squared.
+
+        Parameters
+        ----------
+        y : np.ndarray[nfreq, ntime]
+            Chi-squared per degree of freedom.
+        w : np.ndarray[nfreq, ntime]
+            Inverse variance of the chi-squared per degree of freedom,
+            with zero indicating previously masked samples.
+
+        Returns
+        -------
+        mask : np.ndarray[nfreq]
+            Boolean mask that indicates frequencies and times where
+            chi-squared deviates significantly from the local median.
+        """
+        y = np.ascontiguousarray(y, dtype=np.float64)
+
+        win_size = (self.win_f, self.win_t)
+
+        if not self.estimate_var:
+            mad_y = np.ones_like(y)
+
+        # Slowly reduce the threshold.  At each iteration generate a new estimate
+        # of the background sky and the variance using the current mask.
+        mask = w == 0.0
+        for nsigma in self.threshold:
+
+            f = np.ascontiguousarray(~mask * w, dtype=np.float64)
+
+            # Calculate the local median
+            med_y = weighted_median.moving_weighted_median(y, f, win_size)
+
+            # Calculate the deviation from the median, normalized by the
+            # expected standard deviation
+            dy = (y - med_y) * np.sqrt(w)
+
+            # If requested, estimate the variance in the test statistic
+            # using the median absolute deviation.
+            if self.estimate_var:
+                f = np.ascontiguousarray(f > 0.0, dtype=np.float64)
+                mad_y = 1.48625 * weighted_median.moving_weighted_median(
+                    np.abs(dy), f, win_size
+                )
+
+            # Generate a mask using sumthreshold
+            stmask = rfi.sumthreshold(
+                dy,
+                self.max_m,
+                start_flag=mask,
+                threshold1=nsigma,
+                remove_median=False,
+                correct_for_missing=True,
+                rho=1.0,
```

Comment:
Is rho would be `rho = self.rho` ?

Reply:
Rho determines how the threshold is lowered with each iteration and is used to determine self.threshold in the setup method.  For chi-squared masking and sensitivity masking, I've moved this iteration outside of the sumthreshold function so that we can also update the median and median absolute deviation as we lower the threshold.